lang/cs/Org.Apache.REEF.Common/metrics/MetricsSystem/ImmutableCounter.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Common.Metrics.Api;
+
+namespace Org.Apache.REEF.Common.Metrics.MetricsSystem
+{
+    /// <summary>
+    /// Immutable Counter implementation. This class is not thread safe and since it 
+    /// is readonly thread safety is not required.
+    /// </summary>
+    internal sealed class ImmutableCounter : ImmutableMetricsImpl
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="info">Meta data of the counter</param>
+        /// <param name="value">Value of the counter</param>
+        public ImmutableCounter(IMetricsInfo info, long value)
```

Comment:
So any public member inside an internal class will be internal.
I was reading about this and advantage of writing methods of internal class as public is that at later point of time if we make the class public, we do not need to change individual methods.